src/pysweepme/Ports.py

```diff
@@ -830,6 +875,61 @@ def read_internal(self, digits=0):
         return answer
 
 
+class SOCKETport(Port):
+
+    def __init__(self, ID):
+
+        super(__class__, self).__init__(ID)
+
+    def open_internal(self):
+
+        port_ID = self.port_properties["ID"]
+        ok, HOST, PORT = is_IP(port_ID)
+        if not ok:
+            raise ValueError(f"Port string {[port_ID]} is not a valid IPV4 address")
+
+        self.port = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.port.settimeout(0.1)
+        self.port.connect((HOST, PORT))
+
+        # if self.port_properties["TCPIP_EOLwrite"] is not None:
+        #     self.port.write_termination = self.port_properties["TCPIP_EOLwrite"]
+
+        # if self.port_properties["TCPIP_EOLread"] is not None:
+        #     self.port.read_termination = self.port_properties["TCPIP_EOLread"]
+
+    def close_internal(self):
+        self.port.close()
+
+    def get_identification(self):
+
+        self.write("*IDN?")
+        return self.read()
+
+    def write_internal(self, cmd: str):
+
+        self.port.sendall(cmd.encode('latin-1'))
+        time.sleep(self.port_properties["delay"])
```

Comment:
orignal idea of delay was to make sure to send message no more often than a given time interval. It is sometimes needed for COM ports as some instruments are missing messages when commands are sent too frequently. In case of SOCKET it might not be needed at all as typically no package is lost but rather buffered by the ethernet adapter.